### ISSUE TITLE
`next-hop-aliases` is missing requested/configured hostname

### ISSUE THREAD

**Issue by PiotrSikora** [httpwg/http-extensions#2600]
> The items in the list include all alias names and canonical names received in CNAME records [[DNS](https://httpwg.org/http-extensions/draft-ietf-httpbis-alias-proxy-status.html#DNS)] during the course of resolving the next hop's hostname using DNS, ***not including the original requested hostname itself***.

I understand that the current draft is specifically targeting forward proxies, but as currently described, the `next-hop-aliases` is missing some information when used by reverse proxies, and it's not self-contained anymore (i.e. the requested/configured next hop is not part of the `Proxy-Status` field), making debugging much harder.

> For example, consider a proxy "proxy.example.net" that receives the following records when performing DNS resolution for the next hop "host.example.com":
>
>     host.example.com.           CNAME   tracker.example.com.
>     tracker.example.com.        CNAME   service1.example.com.
>     service1.example.com.       AAAA    2001:db8::1
>
> The proxy could include the following proxy status in its response:
>
>     Proxy-Status: proxy.example.net; next-hop="2001:db8::1";
>         next-hop-aliases="tracker.example.com,service1.example.com"

I find the fact that `host.example.com` is not included here a bit confusing.

While operators or clients connecting directly to `proxy.example.net` might know that it's a forward proxy and could infer that `host.example.com` was the requested hostname, the clients that are a few hops away from it, won't know this, and might be surprised how the `tracker.example.com` and other aliases got in there.

Furthermore, in a similar scenario with a client connecting to the reverse proxy `proxy.example.net` and requesting `www.example.com`, which is configured to use `host.example.com` as the backend, that response is missing the most critical piece of information, and instead of helping, it adds more confusion.

As such, I'd suggest that the requested/configured hostname should be included as the first entry in `next-hop-aliases`, especially when `next-hop` points to the resolved IP address.

cc @mnot @tfpauly

**Comment by tfpauly** [comment]
This is a good point, especially for reverse proxies. The name of the next hop could be in the next-hop parameter itself, but then there is a choice between the address and the name there. 

For forward proxies, the name is known and the address is interesting. For reverse proxies, i imagine the name of the next hop is more useful than the address, but I can see why both could be wanted. In such case, perhaps we can allow the aliases list to include the next hop name if it differs from the authority requested by the client.

**Comment by PiotrSikora** [comment]
> The name of the next hop could be in the next-hop parameter itself, but then there is a choice between the address and the name there.

Yes, that's unfortunate. I'm not sure how we've missed this in RFC 9209.

Perhaps we could allow both and use `next-hop-aliases` for the configured next hop name, and `next-hop` for the resolved IP address... which actually matches what I'm suggesting we do here by including the requested/configured name as the first entry in `next-hop-aliases`.

> For forward proxies, the name is known and the address is interesting.

But that's only true on the first hop, right? The authority could be changed along the way by intermediaries, and then the original client wouldn't know what started the resolution which resulted in the received `next-hop-aliases`. 

> For reverse proxies, i imagine the name of the next hop is more useful than the address, but I can see why both could be wanted. In such case, perhaps we can allow the aliases list to include the next hop name if it differs from the authority requested by the client.

How would the original client know if that happened in a long chain of proxies? The authority requested by the client connecting to `proxy.example.net` is not included in the `Proxy-Status` header, and therefore it's not known to the original client.

>     Proxy-Status: proxy.example.net; next-hop="2001:db8::1";
>         next-hop-aliases="tracker.example.com,service1.example.com"

It's impossible to tell if that entry was generated by a proxy that was configured to connect to `host.example.com`, which also happens to be the authority it received and therefore it's not included in `next-hop-aliases`, or if it was configured to connect to `tracker.example.com`.

Basically, I'm trying to avoid adding ambiguity and guesswork to the debugging process that uses this header.
